src/opendr/perception/multimodal_human_centric/audiovisual_emotion_learner/avlearner.py

```diff
@@ -0,0 +1,450 @@
+                                shuffle=False)
+        else:
+            loader = DataLoader(data.DataWrapper(dataset),
+                                batch_size=self.batch_size,
+                                num_workers=self.n_workers,
+                                pin_memory=self.device == 'cuda',
+                                shuffle=False)
+
+        self.model = self.model.to(self.device)
+        self.model.eval()
+
+        L = torch.nn.CrossEntropyLoss()
+        loss, acc = trainer.val_one_epoch(-1, loader, self.model, L, mode=mode, device=self.device,
+                                          silent=silent, verbose=verbose)
+        if not silent and verbose:
+            print('Loss: {}, Accuracy: {}'.format(loss, acc))
+
+        return {'cross_entropy': loss, 'acc': acc}
+
+    def _process_video(self, video_path, target_time=3.6, input_fps=30, save_frames=15, target_im_size=224):
+        """
+        This function preprocesses input video file for inference
+        Parameters
+        ----------
+        video_path : str
+            path to video file.
+        target_time : float, optional
+            Target time of processed video file in seconds. The default is 3.6.
+        input_fps : int, optional
+            Frames Per Second of input video file. The default is 30.
+        save_frames : int, optional
+            Length of target frame sequence. The default is 15.
+        target_im_size : int, optional
+            Target width and height of each frame. The default is 224.
+
+        Returns
+        -------
+        numpy_video: numpy.array
+                     N frames as numpy array
+
+        """
+        numpy_video = data_utils.preprocess_video(video_path, target_time, input_fps, save_frames,
+                                                  target_im_size, device=self.device)
+        video_scale = 255
+        video_transform = transforms.Compose([
+                              transforms.ToTensor(video_scale)])
+        video = [video_transform(img) for img in numpy_video]
+        video = torch.stack(video, 0).permute(1, 0, 2, 3)
+
+        return video
+
+    def _process_audio(self, audio_path, target_time=3.6, sr=22050):
+        """
+        This function preprocesses an audio file for inference
+
+        Parameters
+        ----------
+        audio_path : str
+            Path to audio file.
+        target_time : int, optional
+            Target duration of audio. The default is 3.6.
+        sr : int, optional
+            Sampling rate of audio. The default is 22050.
+
+        Returns
+        -------
+        y : numpy array
+            audio file saved as numpy array.
+        """
+        y = data_utils.preprocess_audio(audio_path, sr, target_time)
+        mfcc = librosa.feature.mfcc(y, sr=sr, n_mfcc=10)
+        return mfcc
+
+    def load_inference_data(self, audio_path, video_path, target_time=3.6, sr=22050, input_fps=30, save_frames=15,

```

Comment:
In other implementations we tend to incorporate this functionality into the data (using the `open()` function). This is minor though.

Reply:
I am not sure if I understand correctly what you mean, I kept it separate to allow for different preprocessing that might be necessary for a different dataset, so that .infer() simply takes OpenDR data types without any data changes inside

Reply:
This function returns Data-derived objects (Video and Audio) after supplying the path. If this is generic functionality that it is not tied to the specific learner, this could be incorporated into the actual Data class similar to what we have done with images (https://github.com/opendr-eu/opendr/blob/master/src/opendr/engine/data.py#L302). However, if it is tied to this leaner, I think it is better to leave it as it is. I guess that given that this is a multimodal class, it is better in this way.